Makefile.am

```diff
@@ -250,7 +250,19 @@ endif
 
 proxy_ldflags = "-X main.DefaultSocketPath=$(localstatedir)/run/cc-oci-runtime/proxy.sock"
 cc-proxy: $(cc_proxy_sources)
-	$(AM_V_GO)go build -o $@ -ldflags=$(proxy_ldflags) $(srcdir)/proxy
+	[ -n "$$GOPATH" ] && \
+		( \
+			$(AM_V_GO)go build -o $@ -ldflags=$(proxy_ldflags) $(srcdir)/proxy ;\
+		) || \
+		( \
+			export GOPATH=$$(readlink -f $(srcdir))/.gopath ; \
+			rm -rf "$$GOPATH"; \
+			mkdir "$$GOPATH"; \
+			mkdir -p "$$GOPATH"/src/github.com/01org/ ; \
+			ln -sf $$(readlink -f $(srcdir)) "$$GOPATH"/src/github.com/01org/cc-oci-runtime ; \
+			cp -r $(srcdir)/vendor/* "$$GOPATH/src" ; \
+			go build -o $@ -ldflags=$(proxy_ldflags) $(srcdir)/proxy ;\
```

Comment:
This looks a bit complex to my eyes. I suggest we turn this into a `.sh.in` script which gets called by `Makefile.am`. That way:

- You can add comments to explain what is happening, without having them appear in the make output.
- You won't need to use the "double-dollar" syntax which is difficult to read.
- You won't need to use the semi-colons.

Overall, I think a separate script (containing `set -e` at the top) should be easier to maintain and would make the logic clearer.

Ideally, the script would live in a `scripts/` directory. We haven't got one of those yet but this PR could always introduce one (we cant then move the scripts currently in `data/` to `scripts/`).

Please can you make sure that "`make dist` continues to work (it should include the new script).